### ISSUE TITLE
Support dry-run in the admission webhook

### ISSUE THREAD

**Issue by afrittoli** [tektoncd/pipeline#1513]
# Expected Behavior
I can kubectl diff tekton pipeline resources

# Actual Behavior
On the plumbing repo:

```
$ kubectl diff -f tekton/resources/prerelease_checks.yaml
Error from server (InternalError): Internal error occurred: admission webhook "webhook.tekton.dev" does not support dry run
```

**Comment by ImJasonH** [comment]
Good news! This will be very easy to support once we (bad news!) upgrade to the latest knative/pkg and adopt their new style for expressing webhooks.

Basically it boils down to specifying in the ValidatingAdmissionController YAML that validations don't have side effects.

**Comment by vincent-pli** [comment]
@ImJasonH 
I think `knative/pkg` still not support this in it's latest `webhook` implements.

**Comment by ImJasonH** [comment]
The way knative/serving uses knative/pkg, the config is in YAML instead of Go, so it's easier to configure it to support dry-run: https://github.com/knative/serving/blob/master/config/500-webhook-configuration.yaml

**Comment by vincent-pli** [comment]
@ImJasonH 
I see, you are right, they made a `controller` for `admissionregistration`, thanks.

**Comment by vincent-pli** [comment]
Could be close, fix by #1894

**Comment by afrittoli** [comment]
> Could be close, fix by #1894

Nice! This ended up in 0.11.0-rc1 - so as soon and I get that deployed, I'll test this one and close if it works.
